burn-ndarray/Cargo.toml

```diff
@@ -21,18 +21,29 @@ blas-openblas-system = [
     "openblas-src/system",
 ]
 
+# Use randomly generated seed during build. This is useful for testing purposes.
+build-generated-seed = []
+
 [dependencies]
-burn-tensor = { version = "0.5.0", path = "../burn-tensor" }
-blas-src = { version = "0.8.0", default-features = false, optional = true }
-openblas-src = { version = "0.10.5", optional = true }
+blas-src = {version = "0.8.0", default-features = false, optional = true}
+burn-tensor = {version = "0.5.0", path = "../burn-tensor", default-features = false}
+openblas-src = {version = "0.10.5", optional = true}
+
+derive-new = {version = "0.5.9", default-features = false} # Default enables std 
+libm = "0.2.6" # no-std compatible
+ndarray = {version = "0.15.6", default-features = false} # Default enables std 
+
+rand = {version = "0.8.5", default-features = false, features = ["small_rng"]} # Default enables std
 
-ndarray = "0.15.6"
-libm = "0.2.6"
-derive-new = "0.5.9"
-rand = "0.8.5"
-num-traits = "0.2.15"
+num-traits = {version = "0.2.15", default-features = false, features = ["libm"]} # Default enables std
+spin = {version = "0.9.5", features = ["mutex", "spin_mutex"]} # using in place of use std::sync::Mutex;
+
+# Used to generate a seed for random at build time
```

Comment:
What is the use case for generating a seed at build time?

Reply:
When I removed auto-seeding from OS (no_std environment does not have it), the tests broke. Seeding for each test would have required more code changes. So I introduced `build-generated-seed` flag. But now I am thinking if we have `std` flag, which is set by default, then we still can make the tests work. 

It might worth keeping the build time seed. It could be useful in some environments where seeding is not available. 

It's your call really. We can remove it if you'd like.